.github/workflows/downstream-tests-on-comment.yml

```diff
@@ -0,0 +1,183 @@
+      ubuntu-latest-jaspFactor: ${{ steps.update-output.outputs.ubuntu-latest-jaspFactor }} 
+      macOS-latest-jaspDescriptives: ${{ steps.update-output.outputs.macOS-latest-jaspDescriptives }} 
+      macOS-latest-jaspTTests: ${{ steps.update-output.outputs.macOS-latest-jaspTTests }} 
+      macOS-latest-jaspAnova: ${{ steps.update-output.outputs.macOS-latest-jaspAnova }} 
+      macOS-latest-jaspRegression: ${{ steps.update-output.outputs.macOS-latest-jaspRegression }} 
+      macOS-latest-jaspFrequencies: ${{ steps.update-output.outputs.macOS-latest-jaspFrequencies }} 
+      macOS-latest-jaspFactor: ${{ steps.update-output.outputs.macOS-latest-jaspFactor }} 
+      windows-latest-jaspDescriptives: ${{ steps.update-output.outputs.windows-latest-jaspDescriptives }} 
+      windows-latest-jaspTTests: ${{ steps.update-output.outputs.windows-latest-jaspTTests }} 
+      windows-latest-jaspAnova: ${{ steps.update-output.outputs.windows-latest-jaspAnova }} 
+      windows-latest-jaspRegression: ${{ steps.update-output.outputs.windows-latest-jaspRegression }} 
+      windows-latest-jaspFrequencies: ${{ steps.update-output.outputs.windows-latest-jaspFrequencies }} 
+      windows-latest-jaspFactor: ${{ steps.update-output.outputs.windows-latest-jaspFactor }} 
+
+    env:
+      R_REMOTES_NO_ERRORS_FROM_WARNINGS: true
+      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ needs.get-pr-sha.outputs.sha }}
+
+      - uses: r-lib/actions/setup-r@master
+        with:
+         r-version: ${{ matrix.r }}
+
+      - name: clone module
+        run: |
+          echo cloning "https://github.com/${{ matrix.module }}.git"
+          git clone "https://github.com/${{ matrix.module }}.git"
+        shell: bash
+        
+      - uses: jasp-stats/jasp-actions/setup-test-env@master
+
+      - name: Install igraph from source
+        if: ${{ matrix.os }} == 'ubuntu-latest' && ${{ matrix.module }} == 'jasp-stats/jaspFactor'
+        run: install.packages("igraph", type = "source", repos = "https://cloud.r-project.org")
+        shell: Rscript {0}
+
+      - name: Install jaspGraphs
+        run: |
+           print(sprintf("Installing jaspGraphs from commit %s and ref %s", Sys.getenv("GITHUB_SHA", unset = "MISSING SHA!"), Sys.getenv("GITHUB_REF", unset = "MISSING REF!")))
+           remotes::install_local(".")
+        shell: Rscript {0}
+
+      - name: Run unit tests
+        run: |
+          setwd(basename("${{ matrix.module }}"))
+          source("tests/testthat.R")
+        shell: Rscript {0}
+        
+#       - name: fake unit tests for debugging
+#         run: |
+#           if [ ${{ matrix.module }} == 'jasp-stats/jaspDescriptives' ]; then
+#             exit 1
+#           fi
+#         shell: bash
+       
+      - name: set error if tests fail
+        id: update-output
+        if: ${{ failure() }}
```

Comment:
Cool feature and definitely useful for our core packages!

I do wonder if it would be possible to extract more information (i.e., failure reasons) -- could we propagate this from R/logs (really not versed in what's possible in GitHub Actions)?
If a number of repo's are failing for unrelated reasons, it'd still be a bit of a pain to rule those out.

Reply:
> could we propagate this from R/logs (really not versed in what's possible in GitHub Actions)?

In principle, yes. We could do something like this:

1. Have each matrix job (e.g., macOS-jaspAnova) upload an artifact with log files when the unit tests fail.
2. Download and aggregate those logs in the next step.

There are some downsides though. This is quite some work to set up though. Uploading and downloading artifacts is limited in terms of GB / month, so we wouldn't be able to do use these actions too often. I'd assume that the people using this most would be the three of us, and you can just click on the Actions tab at the top of a repo and then see all the jobs (e.g., see [here](https://github.com/vandenman/jaspGraphs/runs/4468682243?check_suite_focus=true)). 

I can include a link directly to the run in the comment if you think that's easier?

Reply:
Ah, for some reason I didn't think those logs would be available. Scratch what I said then.
LGTM.